pkg/kubelet/apis/podresources/server_v1_test.go

```diff
@@ -581,3 +793,33 @@ func equalAllocatableResourcesResponse(respA, respB *podresourcesapi.Allocatable
 	}
 	return equalContainerDevices(respA.Devices, respB.Devices)
 }
+
+func equalGetResponse(ResA, ResB *podresourcesapi.GetPodResourcesResponse) bool {
+	podResA := ResA.PodResources
+	podResB := ResB.PodResources
+	if podResA.Name != podResB.Name {
+		return false
+	}
+	if podResA.Namespace != podResB.Namespace {
+		return false
+	}
+	if len(podResA.Containers) != len(podResB.Containers) {
+		return false
+	}
+	for jdx := 0; jdx < len(podResA.Containers); jdx++ {
+		cntA := podResA.Containers[jdx]
+		cntB := podResB.Containers[jdx]
+
+		if cntA.Name != cntB.Name {
+			return false
+		}
+		if !equalInt64s(cntA.CpuIds, cntB.CpuIds) {
+			return false
+		}
+
+		if !equalContainerDevices(cntA.Devices, cntB.Devices) {
```

Comment:
you seem not to check for dynamic resources here, this could explain why the suspicious tests above (double mock registration, the latter possibly wrong) still passes